src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1039,9 +1039,9 @@ private bool CheckFieldValueKind(SyntaxNode node, BoundFieldAccess fieldAccess,
                         fieldIsStatic == containing.IsStatic &&
                         (fieldIsStatic || fieldAccess.ReceiverOpt.Kind == BoundKind.ThisReference) &&
                         (Compilation.FeatureStrictEnabled
-                            ? TypeSymbol.Equals(fieldSymbol.ContainingType, containing.ContainingType, TypeCompareKind.ConsiderEverything2)
+                            ? TypeSymbol.Equals(fieldSymbol.ContainingType, containing.ContainingType, TypeCompareKind.AllNullableIgnoreOptions)
```

Comment:
>Equals [](http://example.com/codeflow?start=41&length=6)

It looks like there is a somewhat similar logic in ```Binder.AccessingAutoPropertyFromConstructor```. We might want to align it.

Reply:
Sure. Though `AccessingAutoPropertyFromConstructor` appears to be behaving correctly already.